modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -203,22 +225,115 @@ public CompletableFuture<Void> startAsync(ComponentContext componentContext) {
                         inBusyLock(busyLock, () -> onCreateZone(parameters).thenApply((ignored) -> false))
         );
 
-        return nullCompletedFuture();
+        return allOf(processZonesOnStart, processAssignmentsFuture);
+    }
+
+    private CompletableFuture<Void> processZonesOnStart(long recoveryRevision, @Nullable HybridTimestamp lwm) {
+        int earliestCatalogVersion = catalogMgr.activeCatalogVersion(hybridTimestampToLong(lwm));
+        int latestCatalogVersion = catalogMgr.latestCatalogVersion();
+
+        var startedZones = new IntOpenHashSet();
+        var startZoneFutures = new ArrayList<CompletableFuture<?>>();
+
+        for (int ver = latestCatalogVersion; ver >= earliestCatalogVersion; ver--) {
+            int ver0 = ver;
+            catalogMgr.zones(ver).stream()
+                    .filter(tbl -> startedZones.add(tbl.id()))
+                    .forEach(zoneDescriptor -> startZoneFutures.add(
+                            calculateZoneAssignmentsAndCreateReplicationNodes(recoveryRevision, ver0, zoneDescriptor)));
+        }
+
+        return allOf(startZoneFutures.toArray(CompletableFuture[]::new))
+                .whenComplete((unused, throwable) -> {
+                    if (throwable != null) {
+                        LOG.error("Error starting zones", throwable);
+                    } else {
+                        LOG.debug("Zones started successfully");
```

Comment:
I'd also log ms recovery revision, earliest and latest catalog versions and ids of all zones that were started.

Reply:
Added `Zones started successfully [earliestCatalogVersion={}, latestCatalogVersion={}, startedZoneIds={}]`